feat(ui): implement dark mode

kduret opened the pull request:
## Description

Improve CSS file by removing dynamique CSS in order to manage different colors on tab menu. Now menu colors are managed by hardcoded value, so it's useless.
I also add the ability to select a theme in the personal parameters in order to simplify the dark theme development.

thanks @julienmathis 

**Fixes** MON-11932

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Check dark mode switch in my account

julienmathis committed:
empty color files

julienmathis committed:
change header

julienmathis committed:
change header

julienmathis committed:
add header

julienmathis committed:
rename color_css.php into style-color.css

julienmathis committed:
change file format

julienmathis committed:
improve header file, remove old functionnalities, update headers, add first step of theme selection

julienmathis committed:
remove old directory

julienmathis committed:
clean file

julienmathis committed:
Duplicate Theme to init dark theme

julienmathis committed:
improve SQL schema in order to manage theme

julienmathis committed:
clean file

julienmathis committed:
Add possibility to manage theme

kduret committed:
Merge branch 'develop' into improve-css

kduret committed:
linting

jeremyjaouen committed:
add query to get contact_theme

jeremyjaouen committed:
remove dark mode

kduret committed:
update database column

kduret committed:
Merge branch 'improve-css' of https://github.com/centreon/centreon into improve-css

jeremyjaouen committed:
change Centreon-Light for light

jeremyjaouen committed:
Merge branch 'improve-css' of github.com:centreon/centreon into improve-css

kduret committed:
delete template value in ptions table

kduret committed:
Merge branch 'improve-css' of https://github.com/centreon/centreon into improve-css

jeremyjaouen committed:
replace php by css file

jeremyjaouen committed:
Merge branch 'improve-css' of github.com:centreon/centreon into improve-css

kduret committed:
remove useless folder and fix linting

kduret committed:
Merge branch 'improve-css' of https://github.com/centreon/centreon into improve-css

kduret committed:
add carriage return

kduret committed:
Merge branch 'develop' into improve-css

kduret committed:
lint css

kduret committed:
fix clapi test

kduret committed:
fix indent

kduret committed:
remove kb php warning

kduret committed:
fix css issues catched with sonarqube

kduret committed:
linting in htmlheader

kduret committed:
linting

callapa committed:
Delete unnecessary CSS file

callapa commented on www/include/core/header/htmlHeader.php:
```diff
@@ -44,75 +45,140 @@
+$statement->execute();
+if ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
+    switch ($result['contact_theme']) {
+        case 'light':
+            $cssThemes = "Centreon-Light";
+            break;
+        case 'dark':
+            $cssThemes = "Centreon-Dark";
+            break;
+        default:
```
Use the theme name directly.
Be careful, for security reasons, please check that the name follows the following regex: ^([0-9a-zA-Z-]+)$

bdauria committed:
Set centreon-frontend branch to dark mode's one

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

Commit:
Merge branch 'develop' of github.com:centreon/centreon into MON-11932-dark-mode

bdauria committed:
Fix build

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Fix build

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Re-generate package-loc

bdauria committed:
Update centreon-frontend

bdauria committed:
Fix tests

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Re-generate package-lock

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

Noha-ElAbrouki committed:
refactor(header) : replace header scss module with material make styles service component  (#10658)

* Replace Header scss module with material makeStyles serviceComponent

* Replace Header scss module with material makeStyles for service component

* replace header css module for service component

* fix zIndex

Co-authored-by: vagrant <vagrant@localhost.localdomain>

Noha-ElAbrouki committed:
refactor(header) : replace header scss module with material make styles host component  (#10659)

* Replace Header scss module with material makeStyles hostComponent

* replace header scss module with material make styles host component

* replace header scss module with material make styles host component

Co-authored-by: vagrant <vagrant@localhost.localdomain>

Noha-ElAbrouki committed:
Replace Header scss module with material makeStyles hedaerIndex (#10656)

Co-authored-by: vagrant <vagrant@localhost.localdomain>

Noha-ElAbrouki committed:
refactor(header): replace header scss module with material make styles poller menu  (#10662)

* Replace Header scss module with material makeStyles pollerMenu

* replace header scss module with material make styles poller menu

* fix eslint issue

Co-authored-by: vagrant <vagrant@localhost.localdomain>

Noha-ElAbrouki committed:
Manage Header and Footer (#10680)

hyahiaoui-ext committed:
add selected theme to user informations endpoint (#10641)

* add selected theme to user informations endpoint

* updating user informations api doc

* Fix changes theme property in contact entity

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Adapt changes from develop

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

Noha-ElAbrouki committed:
refactor(userMenu): refactor user menu to use standard buttons and links (#10688)

* Refactor User menu to use standard buttons and links

* Refactor User menu to use standard buttons and links

* Refactor User menu to use standard buttons and links

* Refactor User menu to use standard buttons and links

* fix ts rules

* fix ts rules

* Update www/front_src/src/Header/userMenu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* refactor user menu to use standard buttons and links

* Update www/front_src/src/Header/userMenu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Header/userMenu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Header/userMenu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Header/userMenu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* fix conflict and adapt userMenu

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

sc979 committed:
:construction: fix(chore): exclude branch's job from analysis (#10741)

sc979 committed:
Revert ":construction: fix(chore): exclude branch's job from analysis (#10741)"

This reverts commit aeffcb61d4aa227bfc77d734e616c6b0755b891e.

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge branch 'MON-11932-dark-mode' of github.com:centreon/centreon into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Fix build

bdauria committed:
Fix E2E test

bdauria committed:
Fix aria fet

ataghzout committed:
fix(php): fix php warning in header.php (#10767)

Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

Noha-ElAbrouki committed:
enh: centreon dark mode for resources status detail component  (#10716)

* adapt color for detail ResourcesPage

* package-lock

Noha-ElAbrouki committed:
enh: centreon dark mode for resources status-graph component- (#10707)

* adapt color of container header -component detail-

* 11989 adapt color for legend graph

* 11989 adapt color for graph component

* fix color

* fix fontSize of graph

* fix fontSize graph

* package-lock

* :construction: fix(chore): exclude branch's job from analysis (#10741)

* Revert ":construction: fix(chore): exclude branch's job from analysis (#10741)"

This reverts commit 7cd2b34d481a3443002798816c486e4107f9645d.

Co-authored-by: sc979 <34628915+sc979@users.noreply.github.com>
Co-authored-by: schapron <schapron@centreon.com>

kduret committed:
Merge branch 'develop' into MON-11932-dark-mode

bdauria committed:
Remove Module and Widget features

bdauria committed:
Merge branch 'MON-11932-dark-mode' of github.com:centreon/centreon into MON-11932-dark-mode

ataghzout committed:
enh(ui): use css variables for colors (#10671)

* adding color variables file and exporting from 1000 lines -> end in style.css

* Exportation of all variables [Centreon-light]

* fixes missing variables and imports

* Fix all paths

* change primary color legacy

* Move all variables to variables.css

* resolve duplications and conflicts with variables

* change background body color

* fix retours

* Fix linting

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>
Co-authored-by: Laurent Calvet <lcalvet@centreon.com>

ataghzout committed:
Mon 12354 (#10704)

* adding color variables file and exporting from 1000 lines -> end in style.css

* Exportation of all variables [Centreon-light]

* fixes missing variables and imports

* Fix all paths

* change primary color legacy

* Move all variables to variables.css

* resolve duplications and conflicts with variables

* change background body color

* fix retours

* centralizing styles in a generic theme and creating ligh and dark themes

* Implementation of the generic theme

* adding more color variables and fix indentations

* centralizing styles in a generic theme and creating ligh and dark themes

* Implementation of the generic theme

* linking some of varibles to their colors variables

* add all colors of template

* fix import variables

* fix imports

* fix retours

* Fix linting error

* fix errors phpcs

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>
Co-authored-by: Laurent Calvet <lcalvet@centreon.com>

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge branch 'MON-11932-dark-mode' of github.com:centreon/centreon into MON-11932-dark-mode

ataghzout committed:
Centreon Dark mode for Custom Views (#10790)

* adding color variables file and exporting from 1000 lines -> end in style.css

* Exportation of all variables [Centreon-light]

* fixes missing variables and imports

* Fix all paths

* Move all variables to variables.css

* resolve duplications and conflicts with variables

* change background body color

* fix retours

* centralizing styles in a generic theme and creating ligh and dark themes

* Implementation of the generic theme

* adding more color variables and fix indentations

* centralizing styles in a generic theme and creating ligh and dark themes

* Implementation of the generic theme

* linking some of varibles to their colors variables

* add all colors of template

* inversing colors in widget tabs to be blue as active and white as passive (#10719)

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>

* enh(theme): Create dark theme for background, tabs and buttons (#10740)

* add borders to unselected tabs to make them visible

* switching custom view tabs and background to dark mode

* changing tabs colors to match newer versions

* add primary colors

* fix bug variables.css

* switch primary dark to light mode and light to dark mode

* applying dark mode to mini popups in custom view

Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

* Mon 12372 (#10760)

* switching custom view tabs and background to dark mode

* WIP

* add drak mode to widget preferences

* Adaptation & rebase  on MON-12371

* fixing primary dark color and tweaking some styles

* changing checkbox colors to match light and dark mode

* update style of checkbox

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

* delete empty variables.css

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>
Co-authored-by: hyahiaoui-ext <97593234+hyahiaoui-ext@users.noreply.github.com>

hyahiaoui-ext committed:
Centreon legacy components to dark mode (#10793)

* adding color variables file and exporting from 1000 lines -> end in style.css

* Exportation of all variables [Centreon-light]

* fixes missing variables and imports

* Fix all paths

* Move all variables to variables.css

* resolve duplications and conflicts with variables

* change background body color

* fix retours

* centralizing styles in a generic theme and creating ligh and dark themes

* Implementation of the generic theme

* adding more color variables and fix indentations

* centralizing styles in a generic theme and creating ligh and dark themes

* Implementation of the generic theme

* linking some of varibles to their colors variables

* add all colors of template

* inversing colors in widget tabs to be blue as active and white as passive (#10719)

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>

* enh(theme): Create dark theme for background, tabs and buttons (#10740)

* add borders to unselected tabs to make them visible

* switching custom view tabs and background to dark mode

* changing tabs colors to match newer versions

* add primary colors

* fix bug variables.css

* switch primary dark to light mode and light to dark mode

* applying dark mode to mini popups in custom view

Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

* Mon 12372 (#10760)

* switching custom view tabs and background to dark mode

* WIP

* add drak mode to widget preferences

* Adaptation & rebase  on MON-12371

* fixing primary dark color and tweaking some styles

* changing checkbox colors to match light and dark mode

* update style of checkbox

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

* adding color variables file and exporting from 1000 lines -> end in style.css

* Exportation of all variables [Centreon-light]

* fixes missing variables and imports

* Fix all paths

* change primary color legacy

* Move all variables to variables.css

* resolve duplications and conflicts with variables

* change background body color

* centralizing styles in a generic theme and creating ligh and dark themes

* Implementation of the generic theme

* adding more color variables and fix indentations

* linking some of varibles to their colors variables

* enh(theme): Create dark theme for background, tabs and buttons (#10740)

* add borders to unselected tabs to make them visible

* switching custom view tabs and background to dark mode

* changing tabs colors to match newer versions

* add primary colors

* fix bug variables.css

* switch primary dark to light mode and light to dark mode

* applying dark mode to mini popups in custom view

Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

* switching custom view tabs and background to dark mode

* WIP

* add drak mode to widget preferences

* Adaptation & rebase  on MON-12371

* fixing primary dark color and tweaking some styles

* changing checkbox colors to match light and dark mode

* enh(ui): adapt Select2 component to dark mode (#10776)

* exportation of style2.css color variables

* applying dark mode to select2 and changing it to match react

* add dark mode to new selected tags select 2

Co-authored-by: Kevin Duret <kduret@centreon.com>

* adapt radio button to light and dark mode (#10786)

refs: MON-12034

Co-authored-by: yassine JABRI <yassine.jabri@goandev.net>

* Mon 12372 (#10760)

* switching custom view tabs and background to dark mode

* WIP

* add drak mode to widget preferences

* Adaptation & rebase  on MON-12371

* fixing primary dark color and tweaking some styles

* changing checkbox colors to match light and dark mode

* update style of checkbox

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

* Mon 12372 (#10760)

* switching custom view tabs and background to dark mode

* WIP

* add drak mode to widget preferences

* Adaptation & rebase  on MON-12371

* fixing primary dark color and tweaking some styles

* changing checkbox colors to match light and dark mode

* update style of checkbox

Co-authored-by: Yahiaoui Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

* add dark mode to normal select 2

* Update variables.css

* Apply suggestions from code review

Fix retours

Co-authored-by: Kevin Duret <kduret@centreon.com>

* fix pointless variables.css file

Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>
Co-authored-by: ataghzout <97593248+ataghzout@users.noreply.github.com>
Co-authored-by: Kevin Duret <kduret@centreon.com>
Co-authored-by: yayassssine <61753581+yayassssine@users.noreply.github.com>
Co-authored-by: yassine JABRI <yassine.jabri@goandev.net>

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

hyahiaoui-ext committed:
Adding dark mode to Host's form (#10801)

* add dark mode to host's form

* fixing text area and another select input's dark mode

* fix retour variable

bdauria committed:
re-generate package-lock

bdauria committed:
Merge branch 'MON-11932-dark-mode' of github.com:centreon/centreon into MON-11932-dark-mode

Yassir-BenBOUBKER committed:
enh(Extensions): Refacto Extension (code style) and adapt dark mode colors if needed (#10769)

* add the Criterias Query to Filter of Extension.

* Extension Filter complete with Criterias Selector and Search Bar

* Extension add Listing

* Complete Extension and Code Optimisation

* Extension change and Optimisation step 1

* delete comment

* Extension code fixe

* fix labels and disable buttons instead of hiding

* change Updated to UpToDate

* correct labelLicenseNotValid name

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

yayassssine committed:
adabt tables and border to dark mode (#10802)

* adabt tables and border to dark mode
Centreon Dark mode Configuration and Admin listings

* fix return

* fix status color

* change variable (fix return)

Co-authored-by: yassine JABRI <yassine.jabri@goandev.net>

hyahiaoui-ext committed:
updating icons in custom view's widgets (#10811)

* updating icons in custom view widget

* Fix PR change request

Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>

yayassssine committed:
make change in timeline in (menu access) also check user view in (Resources Access)  to dark mode (#10815)

Commit:
Merge branch 'develop' of github.com:centreon/centreon into MON-11932-dark-mode

Commit:
Merge branch 'develop' of github.com:centreon/centreon into MON-11932-dark-mode

Commit:
Fix eslint warning

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Re-generate package-lock

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
re-generate package-lock

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

hyahiaoui-ext committed:
Branch containing all changements for widgets (#10887)

* Add theme to user session for widget usage (#10833)

* add theme to user session for widget usage

* fix list font colors in dark mode

* fix undefined array key user theme

bdauria committed:
Merge branch 'develop' into MON-11932-dark-mode

Yassir-BenBOUBKER committed:
enh(Poller Wizard): refacto - poller Wizard (#10816)

* add shared makeStyle file

* update serverConfigurationWizard on deleting Form-redux

* add jotai Atoms

* delete form redux in poller steps

* fixes

* fix eslint warning

* fixes

* delete form redux in poller steps

* from Classe to function component

* changes and add atom for remoter server data

* title fix

* validation form for pollerStep1

* fixes

* validation form for remoteServerStep1

* add loading button to wizard step2

* change wizard installing page design

* label and models fixed

* simplifie error hundling in form

* apply button loading fix

* delete comment

* fix error handling

* add new label

* add translation to Extension and poller labels

* fixe

* fixes and updates

* use pick in options map

* fixes

* Fixes

* add space

* Conflict Fixe

* fixes

* Update www/front_src/src/Main/useUser.ts

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>
Co-authored-by: Tom Darneix <tomdar87@outlook.com>

Noha-ElAbrouki committed:
consume the theme api (#10890)

Noha-ElAbrouki committed:
refactor(navigation):  left menu (#10830)

* remove old files

* remove old files

* adapt logo for menu

* add sidebar

* add menu for sidebar

* add menuItems

* add arrowIcon component

* add collapsableItems

* add navigationItemSelectedAtom for sidebar

* adapt navigation

* remove gab from menu

* adapt menu

* adapt icons

* adapt style

* adapt style

* adapt style

* update style

* fix align of menu

* adapt style

* adapt style

* fix eslint

* fix eslint

* fix eslint

* add id du sideBar

* adapt style

* refactore names

* updtae addNavigationItemSelected

* update centreon-test-lib version

* remove disabled eslint

* remove eslint-disable-next-line

* update centreon-test-lib

* update align of items

* update addNavigationItemSelected

* update collapsableItems

* updatecollapsableItems

* update slight border

* update style

* upgrade centreon-test-lib

* mutate navigationItemSelected indirectly

* refactore names

* refactore names

* refactore names

* refactore code

* update deleteNavigationItemsSelected

* refactore names

* set items hovered by default

* update searchItemsHoveredByDefault and isItemHoveredByDefault  functions

* reafactore code

* resetSelectedNavigationItems

* refactore code

* refactore code

* refactore code

* refactore code

* Update www/front_src/src/Navigation/Sidebar/Logo/LogoMini/Centreon_Logo_Blanc_C.svg

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Logo/LogoMini/Centreon_Logo_Noir_RVB_C.svg

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* update logo and logoMini

* refactore code

* Update www/front_src/src/Navigation/Sidebar/Menu/index.tsx

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/index.tsx

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

* Update www/front_src/src/Navigation/Sidebar/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/sideBarAtoms.ts

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/sideBarAtoms.ts

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/index.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* unified concept of selectedNavigationItems atom

* refactore and add hoveredNavigationItems

* refactore code

* refactore code

* fixes

* updates

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* fixe eslint

* update memoization

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* refactore ans updates

* updates

* add small elevation to collapsible items list

* Fix items hovered by default

* delete divider from menu

* resize menu

Co-authored-by: Kevin Duret <kduret@centreon.com>
Co-authored-by: Tom Darneix <tomdar87@outlook.com>
Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

Yassir-BenBOUBKER committed:
delete unused js and scss files (#10910)

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

callapa committed:
Merge branch 'develop' into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
enh(chore): Remove sass-loader and unecessary files (#10908)

callapa committed:
enh(SVG): Displays the SVG file in HTML (#10928)

ataghzout committed:
Refactoring dark mode (#10845)

* fixing some dark mode issues and applying new colors

* refactoring all variables hexa & adapt colors

* adapt colors of all legacy component

* change color legend of chart

* changing radio buttons in dashboard pages

* changing leftover checkbox to match the other ones

* applying dark mode to performance's chart title

* changing header title text color

* add dark mode for disabled rows in list

* changing butoon for rules configuration

* change style checkbox service by host

* fixing some dark mode issues and applying new colors

* refactoring all variables hexa & adapt colors

* adapt colors of all legacy component

* change color legend of chart

* changing radio buttons in dashboard pages

* changing leftover checkbox to match the other ones

* applying dark mode to performance's chart title

* changing header title text color

* add dark mode for disabled rows in list

* changing butoon for rules configuration

* change style checkbox service by host

* fix(test): check properly checkbox

* apply dark mode for graphs

* Allows to activate the checkbox by clicking on the parent div (necessary to pass the TA)

Co-authored-by: YAHIAOUI Hamza <hamza.yahiaoui@goandev.net>
Co-authored-by: Ayoub TAGHZOUT <ayoub.taghzout@goandev.net>
Co-authored-by: Kevin Duret <kduret@centreon.com>
Co-authored-by: Laurent Calvet <lcalvet@centreon.com>

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Merge branch 'MON-11932-dark-mode' of github.com:centreon/centreon into MON-11932-dark-mode

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Re-generate package-lock

callapa committed:
Merge remote-tracking branch 'origin' into MON-11932-dark-mode

Commit:
Fix tests

Commit:
Fix eslint error

Commit:
Apply new version of jotai

Commit:
Fix tests

Commit:
Fix eslint error

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

callapa committed:
enh(user): Create an API endpoint to update the user's theme (#10953)

Noha-ElAbrouki committed:
fix(Navigation) :  left menu (#10915)

* fix hoveredNavigationItems for left menu

* add derived atom to avoid code duplication

* Update www/front_src/src/Navigation/Sidebar/sideBarAtoms.ts

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* update ramda import

* update

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/sideBarAtoms.ts

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* Update www/front_src/src/Navigation/Sidebar/Menu/CollapsibleItems.tsx

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

* update import and fix eslint

Co-authored-by: Tom Darneix <tomdar87@outlook.com>

Noha-ElAbrouki committed:
style(Logo):update logo login (#10933)

* update logo for dark mode

* rename image of logo

* add memoization

* fix names

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

Noha-ElAbrouki committed:
feat : Mon 12668 switch mode (#10960)

* add switch mode

* updates

* updates

* implement logic of switch

* update memoization of nac menu

* updates

* consume api

* update memoization for filter of resources

* extract svg

* Update www/front_src/src/Header/switchMode/index.tsx

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

* update names

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

Noha-ElAbrouki committed:
fix(header/themeMode-switch): Make the themeMode switch non-blocking (#10988)

* update switch mode

* Update www/front_src/src/Header/SwitchThemeMode /index.tsx

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

* fixes

Co-authored-by: Bruno d'Auria <bdauria@centreon.com>

Noha-ElAbrouki committed:
replace classnames by clsx (#10987)

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

Noha-ElAbrouki committed:
fix white background displayed during pages navigation when the theme mode is dark (#10991)

bdauria committed:
Merge remote-tracking branch 'origin/develop' into MON-11932-dark-mode

bdauria committed:
Fix spacing in directory

Commit:
Fix build

Commit:
Fix tests

Commit:
Fix tests

hyahiaoui-ext committed:
Replacing png/gif icons with SVGs (#10981)

Replacing most of png/gif icons so they can coop with centreon's current theme.

ataghzout committed:
enh(style): Changing more global style for dark mode implementation (#10946)

Applying more DM to other global styles

callapa committed:
Update composer.lock

callapa committed:
Update composer files
